include/rocksdb/status.h

```diff
@@ -60,6 +60,7 @@ class Status {
     kTryAgain = 13,
     kCompactionTooLarge = 14,
     kColumnFamilyDropped = 15,
+    kInvalidPrecondition = 16,
```

Comment:
I don't know if there are any general rules in rocksdb on adding new error codes vs just adding subcodes (since this seems similar to kInvalidArgument). Maybe someone on the rocksdb team can comment.

We also use invalid argument in other places when transaction state isn't correct (eg. look at `PessimisticTransaction::Prepare` or `PessimisticTransaction::SetName`)

Reply:
I replaced usage of InvalidPrecondition with InvalidArgument